src/components/EntityThumbnailUploader/EntityThumbnailUploader.tsx

```diff
@@ -0,0 +1,175 @@
+
+    for (const entity of thumbnails) {
+      const entityToPatch = entities.find((e) => e.id === entity.id)
+      if (!entityToPatch) continue
+      const thumbnailId = entity.thumbnailId
+      const currentAssignees = entity.users || []
+
+      operations.push({
+        id: entityToPatch.id,
+        projectName: entityToPatch.projectName,
+        data: { updatedAt: newUpdatedAt },
+        currentAssignees,
+      })
+
+      const versionPatch = {
+        productId: entityToPatch.productId,
+        versionUpdatedAt: newUpdatedAt,
+        versionThumbnailId: thumbnailId,
+      }
+
+      versionPatches.push(versionPatch)
+    }
+
+    let productsPatch = patchProductsListWithVersions(versionPatches)
+    try {
+      await updateEntities({ operations, entityType })
+      onUploaded && onUploaded(operations)
+    } catch (error) {
+      productsPatch?.undo()
+    }
+  }
+
+  const inputRef = useRef<HTMLInputElement>(null)
+
+  useEffect(() => {
+      // resetFileUploadState()
+      if (fileUpload) {
+        inputRef.current?.click()
+      }
+  }, [fileUpload])
+
+  const handleFileUpload = async (event: ChangeEvent<HTMLInputElement>) => {
+    const files = event.target.files
+    if (!files) {
+      return
+    }
+
+    const file = files[0]
+    if (!file) {
+      return
+    }
+
+    try {
+      if (!file.type.includes('image')) {
+        throw new Error('File is not an image')
+      }
+
+      let promises = []
+      for (const entity of entities) {
+        const { id, entityType, projectName } = entity
+
+        if (!projectName) throw new Error('Project name is required')
+
+        const promise = axios.post(
+          projectName && `/api/projects/${projectName}/${entityType}s/${id}/thumbnail`,
+          file,
+          {
+            headers: {
+              'Content-Type': file.type,
+            },
+          },
+        )
+
+        promises.push(promise)
+      }
+
+      const res = await Promise.all(promises)
+
+      const updatedEntities = res.map((res, i) => ({
+        thumbnailId: res.data.id as string,
+        id: entities[i].id,
+      }))
+
+      handleThumbnailUpload(updatedEntities)
+    } catch (error: any) {
+      console.error(error)
+    }
+  }
+
+  return (
+    <ThumbnailUploadProvider
+      entities={entities}
+      handleThumbnailUpload={handleThumbnailUpload}
+      inputRef={inputRef}
+    >
+      <Styled.DragAndDropWrapper
+        className={clsx({ isCompact })}
+        onDragEnter={() => setIsDraggingFile(true)}
+      >
+        <ThumbnailWrapper>{children}</ThumbnailWrapper>
+        {isDraggingFile && (
+          <ThumbnailUploader
+            onFinish={handleThumbnailUpload}
+            onDragLeave={() => setIsDraggingFile(false)}
+            onDragOver={(e) => e.preventDefault()}
+            className="thumbnail-uploader"
+            entities={entities}
+          />
+        )}
+        <input type="file" onChange={handleFileUpload} ref={inputRef} />
```

Comment:
When uploading using the input, we should still see the uploading progress like in drag and drop. 


https://github.com/user-attachments/assets/286474b7-79e8-4178-a6bb-0f856a5f7bcb